lib/cli-engine/config-array-factory.js

```diff
@@ -802,7 +802,7 @@ class ConfigArrayFactory {
      */
     _loadExtendedPluginConfig(extendName, ctx) {
         const slashIndex = extendName.lastIndexOf("/");
-        const pluginName = extendName.slice("plugin:".length, slashIndex);
+        const pluginName = slashIndex === -1 ? extendName : extendName.slice("plugin:".length, slashIndex);
```

Comment:
Could we maybe just throw here if `slashIndex === -1`, instead of continuing and trying to load a module with a malformed name (e.g., `"plugin:eslint-config-prettier"`).

`ESLint couldn't find the plugin "plugin:eslint-config-prettier"` could be confusing, because the real issue is a wrong configuration entry in `"extends"`: it's either missing `/config-name` at the end, or has the `plugin:` prefix while it shouldn't have that prefix (in the case that it's actually a config, not a plugin).